app/src/main/java/com/example/randomuserapplication/viewmodel/UserViewModel.kt

```diff
@@ -35,4 +38,16 @@ class UserViewModel @Inject constructor(
         }
     }
 
+    fun setUserId(userId: String) {
+        _selectedUserId.value = userId
+    }
```

Comment:
Why do u have this function to setUserId?

Reply:
The purpose of this function is to ensure that the userId is set to the accurate id of the user being clicked and to be accessed by screen2 via the viewpager in order to retrieve the appropriate user details

Reply:
you can do that without the extra function in the viewmodel, when navigating, pass the user id, get it in the details screen and use it to get UserDetails

Reply:
like safe args?

Reply:
yes

Reply:
I removed unnecessary function and used nav safe args instead.